Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp

```diff
@@ -409,12 +409,85 @@ std::vector<std::string> RGroupDecomposition::getRGroupLabels() const {
   return labels;
 }
 
+void RGroupDecomposition::assignAromaticExplicitHs1(Atom *atom) const {
+  int currentValence = atom->getExplicitValence() - atom->getFormalCharge();
+  const auto &valences =
+      PeriodicTable::getTable()->getValenceList(atom->getAtomicNum());
+  auto closestValenceIt =
+      std::find_if(valences.begin(), valences.end(),
+                   [currentValence](const int allowedValence) {
+                     return (allowedValence >= currentValence);
+                   });
+  if (closestValenceIt != valences.end()) {
+    atom->setNumExplicitHs(*closestValenceIt - currentValence);
+  }
+}
+
+void RGroupDecomposition::assignAromaticExplicitHs2(Atom *atom) const {
```

Comment:
I don't think we should be re-implementing the aromaticity model in the RGD code.
It seems like there's got to be a simpler way to fix the problem than this.